src/components/decks/DeckTable.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+  <div v-if="!isDisabled && (!decks || !decks.length)" class="text-center bg-inexhaustible border border-gray-light my-4 py-8">
+    <h2>No decks found</h2>
+
+    <p>
+      <button @click="$emit('reset-filters')" class="btn btn-blue px-4 py-2">Clear filters</button>
+    </p>
+  </div>
+  <div v-else-if="decks && decks.length">
+    <deck v-for="deck of decks" :key="deck.id" :deck="deck"></deck>
+    <div v-show="haveNextDecks" class="my-4 text-center" ref="scrollLoader">
+      <button class="btn btn-blue py-2 px-4" :disabled="isDisabled" @click="$emit('load-more')">
+        <span v-if="isDisabled">
+          <i class="fas fa-circle-notch fa-spin"></i> Loading...
+        </span>
+        <span v-if="!isDisabled">
+          Load more decks...
+        </span>
+      </button>
+    </div>
+</div>
+  <div v-else class="text-center text-gray text-2xl py-8 px-4">
+    <i class="fas fa-circle-notch fa-spin"></i> Loading...
+  </div>
+</template>
+
+<script>
+import { debounce } from '/src/utils.js'
+import Deck from './Deck.vue'
+
+export default {
+  name: 'DeckTable',
+  props: {
+    isDisabled: Boolean,
+    decks: Array,
+    haveNextDecks: Boolean,
+  },
+  emits: ['reset-filters', 'load-more'],
+  components: {
+    Deck,
+  },
+  mounted () {
+    this.debouncedScrollListener = debounce(this.scrollLoadCheck, 100)
```

Comment:
For the deck listing page, please make it paginated instead of endlessly scrolling (there were a *lot* of decks in the legacy site, so endless scrolling would kind of suck). Optionally, this can just be "next/previous" buttons if calculating out the pages is too complicated with the API return data. I'm not sure how to handle pagination in the query string; maybe don't worry about it for now, since I don't think anyone linked to specific pages ever anyway (given the contents are constantly changing).

Reply:
I've done a simple Previous / Next implementation for now, but adding the pages shouldn't be too hard. I'll look at it this weekend hopefully.